.github/CONTRIBUTING.md

```diff
@@ -8,64 +8,221 @@ When you submit a pull request, a CLA-bot will automatically determine whether y
 
-You can use either one of the following scripts to build Web Chat:
+> Default build flavor is development. Please read [BUILD_SCRIPTS.md](https://github.com/microsoft/BotFramework-WebChat/blob/master/docs/BUILD_SCRIPTS.md) about different build flavors.
 
--  `npm run build` will build once
--  `npm start` will build and continuously rebuild if changes are detected
+There are two ways to build: one-off and continuous build.
 
-## Trying out the build
+- For one-off building, run `npm run build`
+- For continuous building, run `npm start`
 
-Web Chat's playground app under `/packages/` is for testing and debugging.
+The bundle output will be located at:
 
-```sh
-cd packages/playground
-npm run start
-```
+- http://localhost:5000/webchat.js (also `webchat-es5.js` and `webchat-minimal.js`)
+- `packages/bundle/dist/webchat*.js`
 
-Then navigate to http://localhost:3000/, and click on one of the connection options on the upper right corner.
+# Playing the build
 
--  Official MockBot: we hosted a live demo bot for testing Web Chat features
--  Emulator Core: it will connect to http://localhost:5000/v3/directline for [emulated Direct Line service](https://github.com/microsoft/BotFramework-Emulator/tree/master/packages/emulator/cli/)
+There are few ways you can play the build:
 
-You are also advised to test the CDN bundles by copying the test harness from our samples.
+- Playing with playground
+   - `cd packages/playground`
+   - `npm start`
+   - Navigate to http://localhost:3000/, and click on "Official MockBot" on the upper right corner
+- Playing with `webchat-loader`
+   - Navigate to https://compulim.github.io/webchat-loader/
+   - In the version dropdown, select `http://localhost:5000/webchat-es5.js`
+- Write your own web app
+   - `<script src="http://localhost:5000/webchat.js"></script>`
+   - Or using tarballs
 
-## Running integration tests
+# Readying for publication
 
-```bash
-docker-compose up --build --detach
-npm test
-docker-compose down --rmi all
-```
+We care about software quality. It helps prevent regressions, reduce maintenance costs, minimize chores, and enable us to move faster.
 
-## Static code analysis
+Before submitting the pull request, please check your build by running these checks:
 
-Before committing code, please run the following command:
+- [Running automated tests](#running-automated-tests)
+- [Performing static code analysis](#static-code-analysis)
 
-```
-npm run eslint
-```
+## Running automated tests
+
+Automated tests in Web Chat are designed to run inside a stable and isolated test environment. To run the test suite, you will need to [start the test environment](#starting-the-test-environment), then [run the test suite](#running-the-test-suite).
 
-eslint and prettier will ensure that your code follows our linting guidelines without too much effort. If you have any questions about this process, please feel free to leave comments in your Pull Request.
+### Starting the test environment
```

Comment:
Make sure this info matches with `__tests__/README.md`. Might be good to consolidate.

Reply:
Consolidated.